src/MizarData/BaselineAndComparisonHelper.cpp

```diff
@@ -8,53 +8,57 @@
 #include <absl/container/flat_hash_set.h>
 #include <stdint.h>
 
+#include <string>
+
 namespace orbit_mizar_data {
 
 template <typename K, typename V>
 [[nodiscard]] static absl::flat_hash_set<V> ValueSet(const absl::flat_hash_map<K, V>& map) {
   absl::flat_hash_set<V> result;
   std::transform(std::begin(map), std::end(map), std::inserter(result, std::begin(result)),
-                 [](const std::pair<K, V> pair) { return pair.second; });
+                 std::mem_fn(&std::pair<std::add_const_t<K>, V>::second));
   return result;
 }
 
-static absl::flat_hash_map<uint64_t, uint64_t> AddressToSampledFunctionId(
+static absl::flat_hash_map<uint64_t, SFID> AddressToSFID(
     const absl::flat_hash_map<uint64_t, std::string>& address_to_name,
-    const absl::flat_hash_map<std::string, uint64_t>& name_to_frame_id) {
-  absl::flat_hash_map<uint64_t, uint64_t> address_to_frame_id;
+    const absl::flat_hash_map<std::string, SFID>& name_to_sfid) {
+  absl::flat_hash_map<uint64_t, SFID> address_to_sfid;
   for (const auto& [address, name] : address_to_name) {
-    if (const auto it = name_to_frame_id.find(name); it != name_to_frame_id.end()) {
-      address_to_frame_id[address] = it->second;
+    if (const auto it = name_to_sfid.find(name); it != name_to_sfid.end()) {
+      address_to_sfid.try_emplace(address, it->second);
     }
   }
-  return address_to_frame_id;
+  return address_to_sfid;
 }
 
 [[nodiscard]] AddressToIdAndIdToName AssignSampledFunctionIds(
     const absl::flat_hash_map<uint64_t, std::string>& baseline_address_to_name,
     const absl::flat_hash_map<uint64_t, std::string>& comparison_address_to_name) {
-  absl::flat_hash_set<std::string> baseline_names = ValueSet(baseline_address_to_name);
-  absl::flat_hash_set<std::string> comparison_names = ValueSet(comparison_address_to_name);
+  absl::flat_hash_set<std::string> baseline_names =
+      ValueSet<uint64_t, std::string>(baseline_address_to_name);
+  absl::flat_hash_set<std::string> comparison_names =
+      ValueSet<uint64_t, std::string>(comparison_address_to_name);
 
-  absl::flat_hash_map<std::string, uint64_t> name_to_frame_id;
-  absl::flat_hash_map<uint64_t, std::string> frame_id_to_name;
+  absl::flat_hash_map<std::string, SFID> name_to_sfid;
+  absl::flat_hash_map<SFID, std::string> sfid_to_name;
 
-  uint64_t next_frame_id = 1;
+  uint64_t next_sfid_value = 1;
```

Comment:
you could consider implementing operator++ for SFID and use it here instead of uint64_t

Reply:
True. But I thought if SFID as of smth immutable, that is not necessarily consecutive and neither backed with uint64_t. Let's see what other reviewers think.

Reply:
I am not completely sure about this (and probably also not sure about the whole change).
For the concrete situation, I see the argument that SFID is something immutable and does not necessarily have all the properties of natural numbers. Also, if we start adding a ++ function, we quickly also want other numerical operations (+,-,*,/).
If we end up like this, we will have a quite large wrapper to maintain around uint64 just for the purpose of better type safty (which I am in general a big friend of).

I mean what we would really like to have here is a strong typedef, right? I.e. I can only pass in uint64's that I explicitly declared as SFID. Unfortunately this is not really possible.

I'd like to have @beckerhe's opinion on the whole change.

Reply:
Yeah, I am not really sure if this is all worth it :/

Reply:
At least, now we have `flat_hash_map<uint64_t, SFID>`, `flat_hash_map<uint64_t, std::string>` and `flat_hash_map<SFID, std::string>`.
Not `flat_hash_map<uint64_t, uint64_t>`, `flat_hash_map<uint64_t, std::string>` and `flat_hash_map<uint64_t, std::string>`

Reply:
But this could be also archived with a simple typedef, right?

Reply:
True. OTOH, now we have surprising conversions only in some places. With typedefs, these are going to be the same type.

Reply:
> I am not completely sure about this (and probably also not sure about the whole change). For the concrete situation, I see the argument that SFID is something immutable and does not necessarily have all the properties of natural numbers. Also, if we start adding a ++ function, we quickly also want other numerical operations (+,-,*,/). If we end up like this, we will have a quite large wrapper to maintain around uint64 just for the purpose of better type safty (which I am in general a big friend of).
> 
> I mean what we would really like to have here is a strong typedef, right? I.e. I can only pass in uint64's that I explicitly declared as SFID. Unfortunately this is not really possible.
> 
> I'd like to have @beckerhe's opinion on the whole change.

I see it the following way:

- It's great to have SFID as a separate type to make clear it's something else than a natural number
- It would be totally possible to provide operator++ for this type, but then we should make the type properly arithmetic and provide other operations as well.
- But I don't think that's what we want because SFID is still an ID and not a number. It just happens to be based on an integer, but technically it could also use a string as the underlying type.
- That being said, to create new SFIDs it's necessary to understand the underlying type. That's why I like to keep this part as an uint64_t. SFID itself has no notion of a next ID because it doesn't offer a measure of distance or greater smaller comparison. All these things need to be done in the underlying type - which forces the user to understand what they need to do to create a new id.

TLDR: I think I like it that way. It makes sense from a code ownership and responsibility sharing perspective. Personally I would have provided a getter for the value in SFID and then kept the last used SFID around and created a new one like this `SFID{previous_sfid.value() + 1}` but that's really just preference.

Reply:
> So we can't really prevent that conversion.

Wait. I think you accomplished that. A typedef would have the downside of implicit conversions, you approach with the explicit constructor does not. 

My initial comment was really meant as as a suggestion to think about and not as a request to change. I like the argumentation around SFID not being an arithmetic type and I would leave everything as is, or do the suggestion that hebecker had (provide .value()). If you do that, you could also consider removing the `operator std::string`, which is currently the only way of getting the `id_` out of `SFID`.

Reply:
@antonrohr , thank you, my bad. Conversion is prevented, everything's fine. 

I don't really like `.value()` method, as its return type would force us to keep the think uint64_t-based.